Failed to compile with `--with-webrender --enable-rust-debug` on Linux

**Describe the bug**
Failed to compile with `--with-webrender --enable-rust-debug` on Linux

**To Reproduce**

```shell
./configure --with-webrender --enable-rust-debug
```

**Expected behavior**

Finish the `make` process without error.

**OS**

`Linux thinkpadx1extreme 5.13.13-arch1-1 #1 SMP PREEMPT Thu, 26 Aug 2021 19:14:36 +0000 x86_64 GNU/Linux`

**Error callstack**


```
gcc -o temacs.tmp \
  -Demacs  -I. -I. -I../lib -I../lib         -isystem /usr/include/libxml2   -isystem /usr/include/dbus-1.0 -isystem /usr/lib/dbus-1.0/include          -isystem /usr/include/freetype2 -isystem /usr/include/libpng16 -isystem /usr/include/harfbuzz -isystem /usr/include/glib-2.0 -isystem /usr/lib/glib-2.0/include     -MMD -MF deps/.d -MP   -isystem /usr/include/p11-kit-1     -pthread -DEMACS_EXTERN_INLINE -fno-common -Wall -Warith-conversion -Wdate-time -Wdisabled-optimization -Wdouble-promotion -Wduplicated-cond -Wextra -Wformat-signedness -Winit-self -Winvalid-pch -Wlogical-op -Wmissing-declarations -Wmissing-include-dirs -Wmissing-prototypes -Wnested-externs -Wnull-dereference -Wold-style-definition -Wopenmp-simd -Wpacked -Wpointer-arith -Wstrict-prototypes -Wsuggest-attribute=format -Wsuggest-attribute=noreturn -Wsuggest-final-methods -Wsuggest-final-types -Wtrampolines -Wuninitialized -Wunknown-pragmas -Wunused-macros -Wvariadic-macros -Wvector-operation-performance -Wwrite-strings -Warray-bounds=2 -Wattribute-alias=2 -Wformat=2 -Wformat-truncation=2 -Wimplicit-fallthrough=5 -Wshift-overflow=2 -Wvla-larger-than=4031 -Wredundant-decls -Wno-missing-field-initializers -Wno-override-init -Wno-sign-compare -Wno-type-limits -Wno-unused-parameter -Wno-format-nonliteral -g3 -O2   -L.././rust_src/target/debug \
    dispnew.o frame.o scroll.o xdisp.o menu.o  window.o charset.o coding.o category.o ccl.o character.o chartab.o bidi.o cm.o term.o terminal.o xfaces.o   dbusbind.o emacs.o keyboard.o macros.o keymap.o sysdep.o bignum.o buffer.o filelock.o insdel.o marker.o minibuf.o fileio.o dired.o cmds.o casetab.o casefiddle.o indent.o search.o regex-emacs.o undo.o alloc.o pdumper.o data.o doc.o editfns.o callint.o eval.o floatfns.o fns.o font.o print.o lread.o emacs-module.o syntax.o  bytecode.o comp.o dynlib.o process.o gnutls.o callproc.o region-cache.o sound.o timefns.o atimer.o doprnt.o intervals.o textprop.o composite.o xml.o lcms.o inotify.o  profiler.o decompress.o thread.o systhread.o        fontset.o fringe.o image.o  json.o  -lemacsng terminfo.o lastfile.o      ../lib/libgnu.a         -lasound    -lacl    -lrt -ldbus-1       -lxml2  -lgpm   -ltinfo   -lselinux  -lfontconfig -lfreetype     -lgnutls  -lpthread -lanl -llcms2   -lm -lz -ldl -lsystemd  -ljansson  -lgmp  -lxcb  -ldl -lm -lrt -lstdc++
/usr/bin/ld: .././rust_src/target/debug/libemacsng.a(xcb-c1c4e70d0c34a3e3.xcb.ccc38504-cgu.0.rcgu.o): in function `xcb::render::<impl xcb::base::StructPtr<xcb::ffi::render::xcb_render_pictdepth_t>>::visuals':
/home/declan/Source/emacs/rust_src/target/debug/build/xcb-6848f0b72eb79d3b/out/render.rs:387: undefined reference to `xcb_render_pictdepth_visuals_iterator'
/usr/bin/ld: .././rust_src/target/debug/libemacsng.a(xcb-c1c4e70d0c34a3e3.xcb.ccc38504-cgu.0.rcgu.o): in function `xcb::render::<impl xcb::base::StructPtr<xcb::ffi::render::xcb_render_pictscreen_t>>::depths':
/home/declan/Source/emacs/rust_src/target/debug/build/xcb-6848f0b72eb79d3b/out/render.rs:424: undefined reference to `xcb_render_pictscreen_depths_iterator'
/usr/bin/ld: .././rust_src/target/debug/libemacsng.a(xcb-c1c4e70d0c34a3e3.xcb.ccc38504-cgu.0.rcgu.o): in function `xcb::render::<impl xcb::base::Cookie<xcb::ffi::render::xcb_render_query_version_cookie_t>>::get_reply':
/home/declan/Source/emacs/rust_src/target/debug/build/xcb-6848f0b72eb79d3b/out/render.rs:879: undefined reference to `xcb_render_query_version_reply'
/usr/bin/ld: /home/declan/Source/emacs/rust_src/target/debug/build/xcb-6848f0b72eb79d3b/out/render.rs:886: undefined reference to `xcb_render_query_version_reply'
/usr/bin/ld: .././rust_src/target/debug/libemacsng.a(xcb-c1c4e70d0c34a3e3.xcb.ccc38504-cgu.0.rcgu.o): in function `xcb::render::<impl xcb::base::Cookie<xcb::ffi::render::xcb_render_query_pict_formats_cookie_t>>::get_reply':
/home/declan/Source/emacs/rust_src/target/debug/build/xcb-6848f0b72eb79d3b/out/render.rs:956: undefined reference to `xcb_render_query_pict_formats_reply'
/usr/bin/ld: /home/declan/Source/emacs/rust_src/target/debug/build/xcb-6848f0b72eb79d3b/out/render.rs:963: undefined reference to `xcb_render_query_pict_formats_reply'
/usr/bin/ld: .././rust_src/target/debug/libemacsng.a(xcb-c1c4e70d0c34a3e3.xcb.ccc38504-cgu.0.rcgu.o): in function `xcb::render::<impl xcb::base::Reply<xcb::ffi::render::xcb_render_query_pict_formats_reply_t>>::formats':
/home/declan/Source/emacs/rust_src/target/debug/build/xcb-6848f0b72eb79d3b/out/render.rs:1003: undefined reference to `xcb_render_query_pict_formats_formats_iterator'
/usr/bin/ld: .././rust_src/target/debug/libemacsng.a(xcb-c1c4e70d0c34a3e3.xcb.ccc38504-cgu.0.rcgu.o): in function `xcb::render::<impl xcb::base::Reply<xcb::ffi::render::xcb_render_query_pict_formats_reply_t>>::screens':
/home/declan/Source/emacs/rust_src/target/debug/build/xcb-6848f0b72eb79d3b/out/render.rs:1008: undefined reference to `xcb_render_query_pict_formats_screens_iterator'
...(similar output)...
/usr/bin/ld: .././rust_src/target/debug/libemacsng.a(xcb-c1c4e70d0c34a3e3.xcb.ccc38504-cgu.7.rcgu.o): in function `xcb::xfixes::show_cursor':
/home/declan/Source/emacs/rust_src/target/debug/build/xcb-6848f0b72eb79d3b/out/xfixes.rs:1602: undefined reference to `xcb_xfixes_show_cursor'
/usr/bin/ld: .././rust_src/target/debug/libemacsng.a(xcb-c1c4e70d0c34a3e3.xcb.ccc38504-cgu.7.rcgu.o): in function `xcb::xfixes::show_cursor_checked':
/home/declan/Source/emacs/rust_src/target/debug/build/xcb-6848f0b72eb79d3b/out/xfixes.rs:1616: undefined reference to `xcb_xfixes_show_cursor_checked'
/usr/bin/ld: .././rust_src/target/debug/libemacsng.a(xcb-c1c4e70d0c34a3e3.xcb.ccc38504-cgu.7.rcgu.o): in function `xcb::xfixes::create_pointer_barrier':
/home/declan/Source/emacs/rust_src/target/debug/build/xcb-6848f0b72eb79d3b/out/xfixes.rs:1641: undefined reference to `xcb_xfixes_create_pointer_barrier'
/usr/bin/ld: .././rust_src/target/debug/libemacsng.a(xcb-c1c4e70d0c34a3e3.xcb.ccc38504-cgu.7.rcgu.o): in function `xcb::xfixes::create_pointer_barrier_checked':
/home/declan/Source/emacs/rust_src/target/debug/build/xcb-6848f0b72eb79d3b/out/xfixes.rs:1672: undefined reference to `xcb_xfixes_create_pointer_barrier_checked'
/usr/bin/ld: .././rust_src/target/debug/libemacsng.a(xcb-c1c4e70d0c34a3e3.xcb.ccc38504-cgu.7.rcgu.o): in function `xcb::xfixes::delete_pointer_barrier':
/home/declan/Source/emacs/rust_src/target/debug/build/xcb-6848f0b72eb79d3b/out/xfixes.rs:1696: undefined reference to `xcb_xfixes_delete_pointer_barrier'
/usr/bin/ld: .././rust_src/target/debug/libemacsng.a(xcb-c1c4e70d0c34a3e3.xcb.ccc38504-cgu.7.rcgu.o): in function `xcb::xfixes::delete_pointer_barrier_checked':
/home/declan/Source/emacs/rust_src/target/debug/build/xcb-6848f0b72eb79d3b/out/xfixes.rs:1710: undefined reference to `xcb_xfixes_delete_pointer_barrier_checked'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:760: temacs] Error 1
make[1]: Leaving directory '/home/declan/Source/emacs/src'
make: *** [Makefile:454: src] Error 2
```

--- a/configure.ac
+++ b/configure.ac
@@ -1213,6 +1213,13 @@ if test "$enable_silent_rules" = no; then
 else
   AM_DEFAULT_VERBOSITY=0
 fi
+
+AC_ARG_ENABLE(rust-debug,
+[AS_HELP_STRING([--enable-rust-debug],
+[build rust-src in debug mode
+Useful for debugging rust component of emacs-ng.])],
+[ac_enable_rust_debug="${enableval}"],[])
+
 AC_SUBST([AM_DEFAULT_VERBOSITY])
 AC_CONFIG_FILES([src/verbose.mk])
 
@@ -3533,6 +3540,13 @@ if test "${HAVE_X11}" = "yes"; then
   fi
 fi
 
+if test "${window_system}" = "webrender" ; then
+  if test "x$ac_enable_rust_debug" != "x" && test "$opsys" = "gnu-linux"; then
+    XCB_LIBS="-lxcb -lxcb-xfixes -lxcb-shape -lxcb-render"
+    AC_SUBST(XCB_LIBS)
+  fi
+fi
+
 ### Use -lXpm if available, unless '--with-xpm=no'.
 ### mingw32 doesn't use -lXpm, since it loads the library dynamically.
 ### The Cygwin-w32 build uses <noX/xpm.h> instead of <X11/xpm.h>, so
@@ -5737,17 +5751,11 @@ AC_SUBST(CARGO_CLIPPY)
 AC_SUBST(CARGO_RUN)
 AC_SUBST(CARGO_DOC)
 
-AC_ARG_ENABLE(rust-debug,
-[AS_HELP_STRING([--enable-rust-debug],
-[build rust-src in debug mode
-Useful for debugging rust component of Remacs.])],
-[ac_enable_rust_debug="${enableval}"],[])
 if test x$ac_enable_rust_debug != x ; then
 CARGO_FLAGS="$CARGO_FLAGS"
 else
 CARGO_FLAGS="$CARGO_FLAGS --release"
 fi
-
 AC_SUBST(CARGO_FLAGS)
 
 case "${opsys}" in

